### ISSUE TITLE
Breaking change: one action type / action creator

### ISSUE THREAD

**Issue by jamesplease** [jmeas/redux-simple-resource#97]
I'm still working through this, but I want to make RSR the following things:

1. simpler (less surface area API)
2. more powerful
3. more efficient

I think the solution lies in an API like this:

```js
createResources({
  resource: 'cookies',
  resources: [{...}, {...}]
});
```

The idea here is:

1. each CRUD action type is bulk. singular actions are a subset of functionality of bulk
2. one action type for all resources
3. differentiate resources by their name

This would support splitting out the resources among different slices, as you currently do with RSR. But because there are only 4 action types, this will be much less efficient. It would be better to nest them all under a single slice (for instance, `resources`), so that a single reducer is run for them all.

**Comment by jamesplease** [comment]
Another reason for this system is to solve an issue caused by the delete many API.

The custom id attribute is causing problems for delete many. It doesn't really make sense to pass in an array of resources, since that's too much info. So getting the array of IDs from `resources.map(r => r[idAttr])` doesn't work.

For the read one action types, the custom id attribute is used as the key. i.e.;

```
{
  type: 'READ_BOOK',
  bookId: 23
}
```

Since there's no notion of a "plural id" in this lib, the best option, I think, is always passing `ids` as the key. But that's weird. I'll prob do that in 1.1, then this refactor in 2.0 will fix the problemo.

**Comment by jamesplease** [comment]
Resolved in 2.0 branch
